### PR TITLE
Fix polly continue way name

### PR DESCRIPTION
**Pull request by bsudekum**
Closes: https://github.com/mapbox/MapboxNavigation.swift/issues/98

Missing some SSML markup here.

/cc @1ec5

### CODE REVIEW

File: MapboxNavigationUI/RouteVoiceController.swift [hunk 1]
```diff
@@ -242,16 +242,23 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
+    func localizeRoadDescription(_ step: RouteStep, markUpWithSSML: Bool) -> String {
         var road = ""
+        let escapeIfNecessary = {(distance: String) -> String in
+            return markUpWithSSML ? distance.addingXMLEscapes : distance
+        }
         if let name = step.names?.first {
             if let code = step.codes?.first {
-                road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name, code)
+                if markUpWithSSML {
+                    road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "<say-as interpret-as=\"address\">%@</say-as> (<say-as interpret-as=\"address\">%@</say-as>)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name.addingXMLEscapes, code.addingXMLEscapes)
```
Comment (1ec5) [inline MapboxNavigationUI/RouteVoiceController.swift]: Any string provided in the .strings files for a particular localization will override this value.

1. You need to provide distinct string IDs for distinct strings.
1. Maybe this isn’t the right place for this markup. Are you sure you don’t want to surround `name.addingXMLEscapes` in `<say-as>` _before_ inserting it into the format string?

Comment (bsudekum) [inline MapboxNavigationUI/RouteVoiceController.swift]: If it's around `name.addingXMLEscapes ` you get something that sounds like: 

```
Continue on Junipero Freeway opening parenthesis eye two eighty closing parenthesis.
```

Comment (bsudekum) [inline MapboxNavigationUI/RouteVoiceController.swift]: This line cannot have the parens inside the markup https://github.com/mapbox/MapboxNavigation.swift/blob/7aeaff76c175183e3b5b6b75a22d59a811c7b31e/MapboxNavigationUI/RouteVoiceController.swift#L253

Comment (1ec5) [inline MapboxNavigationUI/RouteVoiceController.swift]: This doesn’t work?

```swift
road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), "<say-as interpret-as=\"address\">\(name.addingXMLEscapes)</say-as>", "<say-as interpret-as=\"address\">\(code.addingXMLEscapes)</say-as>")
```

(Split into multiple lines of course.)

It should have the same effect as the code you’ve written, except that now you’d be able to use the same localized format string for both cases and only vary the strings you end up inserting into the format string.

File: MapboxNavigationUI/RouteVoiceController.swift [hunk 2]
```diff
@@ -242,16 +242,23 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
+        let escapeIfNecessary = {(distance: String) -> String in
+            return markUpWithSSML ? distance.addingXMLEscapes : distance
+        }
         if let name = step.names?.first {
             if let code = step.codes?.first {
-                road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name, code)
+                if markUpWithSSML {
+                    road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), "<say-as interpret-as=\"address\">\(name.addingXMLEscapes)</say-as>", "<say-as interpret-as=\"address\">\(code.addingXMLEscapes)</say-as>")
+                } else {
+                    road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name, code)
```
Comment (1ec5) [inline MapboxNavigationUI/RouteVoiceController.swift]: Factor out two local variables, `markedUpName` and `markedUpCode`, that contain the two strings you intend to insert into the `NAME_AND_REF` format string, which will have different values depending on `markUpWithSSML`. Then call `String.localizedStringWithFormat()` just once, passing in `markedUpName` and `markedUpCode`.

File: MapboxNavigationUI/RouteVoiceController.swift [hunk 3]
```diff
@@ -242,16 +242,21 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
...
-    func localizeRoadDescription(_ step: RouteStep) -> String {
+    func localizeRoadDescription(_ step: RouteStep, markUpWithSSML: Bool) -> String {
         var road = ""
+        let escapeIfNecessary = {(distance: String) -> String in
+            return markUpWithSSML ? distance.addingXMLEscapes : distance
+        }
         if let name = step.names?.first {
             if let code = step.codes?.first {
-                road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name, code)
+                let markedUpName = markUpWithSSML ? "<say-as interpret-as=\"address\">\(name.addingXMLEscapes)</say-as>" : name
```
Comment (1ec5) [inline MapboxNavigationUI/RouteVoiceController.swift]: You’ve probably noticed some parallel between this line and `escapeIfNecessary` above. 😉

**Pull request merged**